presto-main/src/main/java/io/prestosql/security/StatementAccessControlManager.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.security;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.log.Logger;
+import io.prestosql.spi.StatementRewriteContext;
+import io.prestosql.spi.security.AccessDeniedException;
+import io.prestosql.spi.security.StatementAccessControl;
+import io.prestosql.spi.security.StatementAccessControlFactory;
+
+import javax.inject.Inject;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static io.prestosql.util.PropertiesUtil.loadProperties;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class StatementAccessControlManager
+        implements StatementAccessControl
```

Comment:
`EventListenerManager` doesn't implements `EventListener`. Could you model it similarly?

Reply:
Avoided StatementAccessControlManager to implement StatementAccessControl